vignettes/connecting_to_sql.Rmd

```diff
@@ -0,0 +1,80 @@
+
+1. executing a separate SQL script from R
+2. writing strings of SQL code in your R script
+3. using [dbplyr](https://dbplyr.tidyverse.org/) to query a database using R code
+
+Which you use will depend on how comfortable with SQL and R and also if you already have existing SQL scripts that you want to execute or you’re writing new database queries. This vignette focuses on that first example, using the `get_clean_sql()` function to read in a separate SQL script and execute from R.
+
+For more information on the other two methods, or on troubleshooting the connection between R and SQL in the Department for Education (DfE), please see the [Interacting with a SQL database section of our Analysts' Guide](https://dfe-analytical-services.github.io/analysts-guide/learning-development/r.html#interacting-with-a-sql-database-from-within-r-scripts).
+
+## Connecting to a database
+
+Usually in the DfE we use a combination of [odbc](https://odbc.r-dbi.org/) and [DBI](https://dbi.r-dbi.org/) to connect to our SQL databases.
+
+How you connect will vary depending on whether you're running R code on your laptop, or as a part of a deployed R Shiny, for running code on your laptop you can automatically use your Windows login (a trusted connection) to grant you access to the database, as the package can automatically detect your user details.
+
+```{r local connection, eval=FALSE}
+# Library calls ====
+
+library(odbc)
+library(DBI)
+
+# Database connection ====
+
+con <- DBI::dbConnect(odbc::odbc(),
+  Driver = "ODBC Driver 17 for SQL Server",
+  Server = "server_name",
+  Database = "database_name",
+  UID = "",
+  PWD = "",
+  Trusted_Connection = "Yes"
+)
+```
+
+For advice on finding your database details, or connecting to a SQL database from an R Shiny app that is deployed on a server, please contact the [Statistics Development Team](mailto:statistics.development@education.gov.uk) who will be able to advise on the setup and steps required.
+
+## Reading a SQL script into R
+
+There are a number of standard characters found in SQL scripts that can cause issues when reading in a SQL script within R, we have created the `get_clean_sql()` function to assist with this. Assume you have connected to the database and assigned that connection to a `con` variable, you would then use the following line to read a cleaned version of your SQL script into R.
+
+```{r reading clean sql, eval=FALSE}
+sql_query <- dfeR::get_clean_sql("path_to_sql_file.sql")
+```
+
+## Executing the SQL query
+
+Now that the SQL query is saved as a variable in the R environment you can pass that into a function to execute against the database. There's a number of potential ways to do this, though a common way is to use `dbGetQuery()` from the [DBI package](https://dbi.r-dbi.org/), setting the statement as your cleaned SQL query.
+
+```{r executing sql query, eval=FALSE}
+sql_query_result <- DBI::dbGetQuery(con, statement = sql_query)
+```
+
```

Comment:
Are there any common things that won't get cleaned by the script and won't run with `dbGetQuery()`?
e.g. will something like the following run:
```
USE [SWFC_Project_Restricted]
GO
```

Worth having a section here to collate those potential trip-hazards.

Reply:
Resolved(ish) by specifying that these are only intended to work with 'SELECT' style statements, with the aim of expanding on this as we go (expecting there'll be additions we can make for anything around connecting to Databricks, and other servers as more people start using it)